### PR TITLE
some Mandatory ChangeZone

### PR DESCRIPTION
**Pull request by Northmoc**

### CODE REVIEW

**Comment by Hanmac** [comment]
what is the reason behind "Mandatory" ? can you still fail to find?

**Comment by tool4ever** [comment]
Yea that's against the rules:
> 701.19b. If a player is searching a hidden zone for cards with a stated quality, such as a card with a certain card type or color, that player isn't required to find some or all of those cards even if they're present in that zone.

File: forge-gui/res/cardsfolder/n/night_terrors.txt [hunk 1]
```diff
@@ -1,5 +1,6 @@
 Name:Night Terrors
 ManaCost:2 B
 Types:Sorcery
-A:SP$ ChangeZone | Cost$ 2 B | Origin$ Hand | Destination$ Exile | ValidTgts$ Player | DefinedPlayer$ Targeted | Chooser$ You | ChangeType$ Card.nonLand | ChangeNum$ 1 | IsCurse$ True | Mandatory$ True | SpellDescription$ Target player reveals their hand. You choose a nonland card from it. Exile that card.
+A:SP$ RevealHand | Defined$ Targeted | ValidTgts$ Player | SubAbility$ ExileCard | StackDescription$ SpellDescription | SpellDescription$ Target player reveals their hand.
+SVar:ExileCard:DB$ ChangeZone | Origin$ Hand | Destination$ Exile | DefinedPlayer$ Targeted | Chooser$ You | ChangeType$ Card.nonLand | ChangeNum$ 1 | IsCurse$ True | StackDescription$ {p:You} chooses a nonland card from it. Exile that card. | SpellDescription$ You choose a nonland card from it. Exile that card.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/n/night_terrors.txt:5]: This still needs Mandatory?

Comment (Hanmac) [reply forge-gui/res/cardsfolder/n/night_terrors.txt:5]: i wonder if there is an Param for revealing hand in the ChangeZone Effect

Comment (Northmoc) [reply forge-gui/res/cardsfolder/n/night_terrors.txt:5]: Isn't a zone still hidden if it has been revealed? So you aren't required to find in this case?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/n/night_terrors.txt:5]: This effect doesn't do any "searching" ⚠️

**Review by tool4ever — APPROVED** [review]

**Pull request merged**